src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -515,6 +515,40 @@ public bool HasBuildSubmissionLoggedErrors(int submissionId)
             return _buildSubmissionIdsThatHaveLoggedErrors?.Contains(submissionId) == true;
         }
 
+        /// <summary>
+        /// Returns a hashset of warnings to be logged as errors for the specified project instance ID.
+        /// Note that WarningsAsMessages takes priority over WarningsAsErrors and are excluded from the set.
+        ///
+        /// If all warnings to be treated as errors should also be treated as messages, return null otherwise
+        /// all warnings will be treated as errors. <see cref="RequestBuilder.ConfigureWarningsAsErrorsAndMessages()"/>
+        /// </summary>
+        /// <param name="context">The build context through which warnings will be logged as errors.</param>
+        /// <returns>A Hashset containing warning codes that should be treated as errors.</returns>
+        public HashSet<string> GetWarningsToBeLoggedAsErrorsByProject(BuildEventContext context)
+        {
+            if(_warningsAsErrorsByProject == null)
+            {
+                return null;
```

Comment:
Returning null for methods that return collections is unexpected. If you change the return type to `ICollection<string>` you'll be able to also return `Array.Empty<string>()` which is preferable.

Reply:
I've changed WarningsAsErrors to generally be of type ICollection, but returning an empty array would change the implementation of warnings as errors. If we want to do that, we should do it in another PR.